skbeam/core/roi.py

```diff
@@ -430,18 +476,27 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
     ring_averages : array
         Radial average of the image. shape is (nx, ).
     """
-    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size)
-
-    bin_edges, sums, counts = utils.bin_1D(np.ravel(radial_val),
-                                           np.ravel(image), nx,
+    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size) 
+    if mask is not None:  
+        mask = np.array( mask, dtype = bool)
+        binr = radial_val[mask]
+        image_mask =     np.array( image )[mask]        
+    else:        
```

Comment:
That's what I mean with "np.asarray(mask,bool)" which should have been np.array(mask,dtype=bool), sorry. I agree with you though, the bool is probably faster.
Anyway, I'm just trying to argue that there doesn't need to be an else statement here. Just if there's a mask, do something extra to the array, if not, leave it alone and leave the rest of the code alone. We shouldn't slow down the code that's been used by others without mask if we can, I think.